src/test/headers_sync_chainwork_tests.cpp

```diff
@@ -85,7 +85,7 @@ BOOST_AUTO_TEST_CASE(headers_sync_state)
             Params().GenesisBlock().nVersion, Params().GenesisBlock().nTime,
             ArithToUint256(1), Params().GenesisBlock().nBits);
 
-    const CBlockIndex* chain_start = WITH_LOCK(::cs_main, return m_node.chainman->m_blockman.LookupBlockIndex(Params().GenesisBlock().GetHash()));
+    const CBlockIndex& chain_start = {*Assert(WITH_LOCK(::cs_main, return m_node.chainman->m_blockman.LookupBlockIndex(Params().GenesisBlock().GetHash())))};
```

Comment:
It's a bit weird to have `Assert` outside, how about:
```suggestion
    const CBlockIndex& chain_start = WITH_LOCK(::cs_main, return *Assert(m_node.chainman->m_blockman.LookupBlockIndex(Params().GenesisBlock().GetHash())));
```

Reply:
Returning a pointer from an anonymous function with no explicit return type seems safer and easier to follow than returning a reference? Not sure what's weird about having the Assert outside the lock. We have a similar instance already:

https://github.com/bitcoin/bitcoin/blob/eb137184482cea8a2a7cdfcd3c73ec6ae5fdd0d6/src/node/blockstorage.cpp#L935

Reply:
I prefer validating closely to the source, but I don't mind either way, thanks for the comment

Reply:
Thanks for catching it and sorry for not specifying in my push comment!

I dropped it on purpose because it caused a compilation warning:

```
[ 73%] Building CXX object src/test/CMakeFiles/test_bitcoin.dir/headers_sync_chainwork_tests.cpp.o
/home/daniela/Developer/BitcoinCore/bitcoin/src/test/headers_sync_chainwork_tests.cpp:55:37: warning: predefined identifier is only valid inside function [-Wpredefined-identifier-outside-function]
   55 |     const CBlockIndex& chain_start{*Assert(WITH_LOCK(::cs_main, return m_node.chainman->m_blockman.LookupBlockIndex(genesis.GetHash())))};
      |                                     ^
/home/daniela/Developer/BitcoinCore/bitcoin/src/util/check.h:115:75: note: expanded from macro 'Assert'
  115 | #define Assert(val) inline_assertion_check<true>(val, __FILE__, __LINE__, __func__, #val)
      |                                                                           ^
```
This didn't happen before my rebase onto #32579, as chain_start was initialized inside a function.

I also had a preference for using the Assert, but I'm not sure if it can still be done somehow... I'm open to suggestions :)

Reply:
Can you try my originally [suggestion](https://github.com/bitcoin/bitcoin/pull/32740#discussion_r2226981997) to put the `Assert` inside?
```C++
const CBlockIndex& chain_start{WITH_LOCK(::cs_main, return *Assert(m_node.chainman->m_blockman.LookupBlockIndex(genesis.GetHash())))};
```
it works for me locally. Or ignore, since we've already acked.